pandas_datareader/oandarest.py

```diff
@@ -0,0 +1,303 @@
+                "granularity": granularity,
+                "from": current_start_rfc3339,
+                "to": current_end_rfc3339,
+                "price": candleFormat,
+                "smooth": "false",
+                "includeFirst": "true" if includeCandleOnStart else "false",
+                "dailyAlignment": "17",
+                "alignmentTimezone": "America/New_York",
+                "weeklyAlignment": "Friday"
+            }
+
+            # print(params)
+
+            try:
+                request = instruments.InstrumentsCandles(
+                    instrument=currencyPair,
+                    params=params)
+
+                response = oanda.request(request)
+                current_start = current_end
+                includeCandleOnStart = False
+            except Exception as error:
+                isExceedResultsLimitError = re.findall(
+                    "The results returned satisfying the query exceeds the maximum size", str(error))
+                isExceedResultsLimitError = True if len(
+                    isExceedResultsLimitError) else False
+
+                if isExceedResultsLimitError:
+                    # Problem: OANDA caps returned range to 5000 results max
+                    # Solution: Reduce requested date interval to return less
+                    # than 5000 results
+                    current_duration /= 2
+                    continue
+                else:
+                    print("ERROR OANDA: " + str(error))
+                    print(str(error))
+                    raise error
+
+            # print(response)
+
+            if not response:
+                continue
+
+            candles = response['candles']
+
+            if not candles:
+                continue
+
+            # print(candles)
+
+            ohlc = ['o', 'h', 'l', 'c']
+            df = pd.io.json.json_normalize(
+                candles,
+                meta=['time', 'volume', 'complete', [
+                    'mid', ohlc], ['ask', ohlc], ['bid', ohlc]]
+            )
+            df['time'] = pd.to_datetime(df['time'])
+
+            with pd.option_context('display.max_columns', None):
+                pass  # print(df)
+
+            dfs.append(df)
+
+        df = pd.concat(dfs)
+
+        df.rename(inplace=True,
+                  index=str,
+                  columns={
+                      "time": "date",
+                      "mid.o": "mid.open",
+                      "mid.h": "mid.high",
+                      "mid.l": "mid.low",
+                      "mid.c": "mid.close",
+                      "ask.o": "ask.open",
+                      "ask.h": "ask.high",
+                      "ask.l": "ask.low",
+                      "ask.c": "ask.close",
+                      "bid.o": "bid.open",
+                      "bid.h": "bid.high",
+                      "bid.l": "bid.low",
+                      "bid.c": "bid.close",
+                  })
+
+        df = df.set_index('date')
```

Comment:
A DataFrame with a MultiIndex as columns will be a great feature to have
http://pandas.pydata.org/pandas-docs/stable/advanced.html

Most other DataReader with candlestick data are using; `Open`, `High`, `Low`, `Close` column name (not `open`, `high`, `low`, `close`)

Here is an example to show how a DataFrame with MultiIndexed columns works
```python
In: periods = 10
In: idx = pd.date_range("2010-01-01", freq="D", periods=10)
In: columns = [np.array(['Mid', 'Mid', 'Mid', 'Mid', 'Ask', 'Ask', 'Ask', 'Ask', 'Bid', 'Bid', 'Bid', 'Bid']),
              np.array(['Open', 'High', 'Low', 'Close', 'Open', 'High', 'Low', 'Close', 'Open', 'High', 'Low', 'Close'])]

In: df = pd.DataFrame(np.random.randn(periods, 12), columns=columns, index=idx)

In: df
Out[59]: 
                 Mid                                     Ask            \
                Open      High       Low     Close      Open      High   
2010-01-01 -0.357908  0.638564 -0.808755  0.029079  0.150620  0.282146   
2010-01-02  1.015372 -0.624254  0.353841  1.491935 -0.322171  0.353117   
2010-01-03  0.051334  0.516878 -1.769581  0.812352 -0.528826  0.677380   
2010-01-04 -1.681174  0.411583 -1.032651  1.924580  0.162397 -0.422782   
2010-01-05  0.512275 -0.112695 -0.705641  0.715283  0.446572 -1.428909   
2010-01-06 -1.197475 -0.025064 -0.997523  1.252269  0.217094 -1.086155   
2010-01-07 -0.935145 -0.446475 -1.220699 -1.135570  0.369933  1.695358   
2010-01-08 -0.259295  0.164924  0.371293 -0.136309  2.280143 -1.256588   
2010-01-09 -0.392548  1.762788 -1.656464  1.548067 -0.333061  0.226948   
2010-01-10 -0.431441  0.674132 -1.381039  0.892813  0.793012 -0.755067   

                                     Bid                                
                 Low     Close      Open      High       Low     Close  
2010-01-01 -1.313252 -1.179413  1.873345 -0.719676 -2.595192  1.182884  
2010-01-02 -0.670126  0.421320  0.318873 -0.756884  1.189023 -0.785939  
2010-01-03 -1.027978  0.609855  0.877085 -0.767293 -1.859567  0.662125  
2010-01-04  1.049496 -1.062435  0.577158 -0.068621 -0.080246 -0.786441  
2010-01-05 -3.193001 -0.540472  2.350964 -1.983940  2.278458 -0.783962  
2010-01-06 -0.101567 -0.145842 -1.600600  0.098544  0.133307  0.262338  
2010-01-07 -0.263570 -1.012718 -0.579255  1.360744  1.661800 -2.185659  
2010-01-08 -2.260959 -1.143676 -0.900669  0.721042 -0.421265 -0.716886  
2010-01-09 -1.229502 -0.442861 -0.767872  0.734038 -0.403405  0.046703  
2010-01-10  1.522349  0.463781  0.522694 -0.700728 -0.726378 -1.144720 

df["Ask"]
Out[60]: 
                Open      High       Low     Close
2010-01-01  0.150620  0.282146 -1.313252 -1.179413
2010-01-02 -0.322171  0.353117 -0.670126  0.421320
2010-01-03 -0.528826  0.677380 -1.027978  0.609855
2010-01-04  0.162397 -0.422782  1.049496 -1.062435
2010-01-05  0.446572 -1.428909 -3.193001 -0.540472
2010-01-06  0.217094 -1.086155 -0.101567 -0.145842
2010-01-07  0.369933  1.695358 -0.263570 -1.012718
2010-01-08  2.280143 -1.256588 -2.260959 -1.143676
2010-01-09 -0.333061  0.226948 -1.229502 -0.442861
2010-01-10  0.793012 -0.755067  1.522349  0.463781

df["Ask"]["Open"]
Out[61]: 
2010-01-01    0.150620
2010-01-02   -0.322171
2010-01-03   -0.528826
2010-01-04    0.162397
2010-01-05    0.446572
2010-01-06    0.217094
2010-01-07    0.369933
2010-01-08    2.280143
2010-01-09   -0.333061
2010-01-10    0.793012
Freq: D, Name: Open, dtype: float64
```